generator/src/main/java/com/linkedin/pegasus/generator/TemplateSpecGenerator.java

```diff
@@ -418,47 +425,46 @@ else if (schema instanceof PrimitiveDataSchema)
    */
   private CustomClasses getCustomClasses(DataSchema schema)
   {
+    return getCustomClasses(schema, _customTypeLanguage);
+  }
+
+  public static CustomClasses getCustomClasses(DataSchema schema, String customTypeLanguage)
+  {
     CustomClasses customClasses = null;
     final Map<String, Object> properties = schema.getProperties();
-    final Object java = properties.get(JAVA_PROPERTY);
-    if (java != null)
-    {
-      if (java.getClass() != DataMap.class)
-      {
-        throw new IllegalArgumentException(schema + " has \"java\" property that is not a DataMap");
-      }
-      final DataMap map = (DataMap) java;
-      final Object custom = map.get(CLASS_PROPERTY);
-      if (custom != null)
-      {
-        if (custom.getClass() != String.class)
-        {
-          throw new IllegalArgumentException(schema + " has \"java\" property with \"class\" that is not a string");
-        }
-        // a custom class specification has been found
-        customClasses = new CustomClasses();
-        customClasses.customClass = new ClassTemplateSpec();
-        customClasses.customClass.setFullName((String) custom);
-        if (!allowCustomClass(schema))
-        {
-          throw new IllegalArgumentException(schema + " cannot have custom class binding");
+    if (customTypeLanguage != null) {
+      final Object java = properties.get(customTypeLanguage);
+      if (java != null) {
+        if (java.getClass() != DataMap.class) {
+          throw new IllegalArgumentException(schema + " has \"" + customTypeLanguage + "\" property that is not a DataMap");
         }
-      }
-      // check for coercer class
-      final Object coercerClass = map.get(COERCER_CLASS_PROPERTY);
-      if (coercerClass != null)
-      {
-        if (coercerClass.getClass() != String.class)
-        {
-          throw new IllegalArgumentException(schema + " has \"java\" property with \"coercerClass\" that is not a string");
+        final DataMap map = (DataMap) java;
+        final Object custom = map.get(CLASS_PROPERTY);
+        if (custom != null) {
+          if (custom.getClass() != String.class) {
+            throw new IllegalArgumentException(schema + " has \"" + customTypeLanguage + "\" property with \"class\" that is not a string");
+          }
+          // a custom class specification has been found
+          customClasses = new CustomClasses();
```

Comment:
This is my major concern for this pull request. Basically current TemplateSpecGeneration implementation introduced such an issue, that is, introduced this language-coupling interpretation of CustomInfoSpec. Instead, at template spec layer, we should only present all the meta information of CustomInfo specified in the PDSC instead of directly interpreting that as Java Class, and as for how to interpret such CustomInfoSpec in code generation, it should leave to individual language code generator to decide. We are in the process of fixing this issue, and should have a RB to commit in the next month. If we make this pull request into master branch, when we later on committed the correct approach of handling custom information in TemplateSpecGenerator, we will have to remove your newly added constructor and bring in a backwards-incompatible change that may break your current usage. If you are ok with this (your code may be broken later), we can pull this in for now.